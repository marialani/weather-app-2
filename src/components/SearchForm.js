import React from "react";
import PropTypes from "prop-types";
import "../styles/SearchForm.css";

const SearchForm = ({ searchText, setSearchText, onSubmit }) => {
  const handleInputChange = (event) => setSearchText(event.target.value);
  const handleSearch = (event) => {
    event.preventDefault();
    onSubmit();
  };
  return (
    <div className="search-form">
      <form>
        <input
          type="text"
          placeholder="Search city..."
          name="search"
          value={searchText}
          onSubmit={handleSearch}
          onChange={handleInputChange}
        />
        <button className="searchButton" type="submit" onClick={handleSearch}>
          SEARCH
        </button>
      </form>
    </div>
  );
};
SearchForm.propTypes = {
  searchText: PropTypes.string.isRequired,
  setSearchText: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
};
export default SearchForm;
