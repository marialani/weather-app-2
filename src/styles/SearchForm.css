.searchForm {
    display: flex;
    justify-content: center;
}

.searchButton {
    padding: 12px 29px;
    margin: 0.5rem;
    background-color: white;
    color: black;
    border: 2px solid #002E45;
    border-radius: 6px;
    font-weight: bold;
    text-align: center;
    box-shadow: rgba(0, 0, 0, 0.4) 0px 2px 4px, rgba(0, 0, 0, 0.3) 0px 7px 13px -3px, rgba(0, 0, 0, 0.2) 0px -3px 0px inset;
}
.searchButton:hover {
cursor: pointer;
color: #2B7994;
border: 2px solid #2B7994;


}
input {
    width: 25%;
    padding: 12px 20px;
    border-radius: 6px;
    border: 2px solid #002E45;
    font-size: medium;
    box-shadow: rgba(0, 0, 0, 0.4) 0px 2px 4px, rgba(0, 0, 0, 0.3) 0px 7px 13px -3px, rgba(0, 0, 0, 0.2) 0px -3px 0px inset;

}
input:focus {
    color: #2B7994;
}