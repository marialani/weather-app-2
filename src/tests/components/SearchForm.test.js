import React from "react";
import { getByRole, render } from "@testing-library/react";
import SearchForm from "../../components/SearchForm";

describe("SearchForm", () => {
  const validProps = {
    searchText: "",
    setSearchText: () => {},
    onSubmit: () => {},
  };
  it("renders correctly", () => {
    const { asFragment } = render(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.onSubmit}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("displays a search button", () => {
    const { getByRole } = render(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.onSubmit}
      />
    );
    expect(getByRole("button")).toHaveTextContent("SEARCH");
  });
});
