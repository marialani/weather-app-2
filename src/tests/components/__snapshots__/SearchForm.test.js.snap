// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search-form"
  >
    <form>
      <input
        name="search"
        placeholder="Search city..."
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
</DocumentFragment>
`;
